### ISSUE TITLE
Checkbox: High contrast mouseover state and checked state

### ISSUE THREAD

**Issue by lynamemi** [OfficeDev/office-ui-fabric-react#4729]
#### Expected behavior:

Checkbox should have a Highlight border-color and text color on hover.

Checked state should be a Highlight fill color with a Window text color. On checked + hover, the fill should go away and leave the border and text Highlight color.

Focus will come into play with this one as hover and focus share a lot of styles.

**Comment by lynamemi** [comment]
Some of this has changed in the 6.0 branch. Waiting until that is out of beta to address this.
